SIS_import/sis_set_url.rb

```diff
@@ -140,28 +329,33 @@ def processTermCourses(mPathwayTermSet,esbToken, outputFile)
 						if (key=="id")
 							courseId = value
 							sections_data = Canvas_API_call("#{$canvasUrl}/api/v1/courses/#{courseId}/sections",
-							                           $canvasToken)
-							#outputFile.write("#{sections_data}\n")
+																							{:per_page => $page_size},
+																							$canvasToken,
+																							nil,
+																							outputFile)
 							sections_data.each { |section|
-								# section is a hash
+								# section is a hash, we will get the sis_section_id value
+								# initialize sectionParsedSISID
+								sectionParsedSISID = nil
 								section.each do |sectionKey, sectionValue|
-									if (sectionKey=="id")
-										section_data = Canvas_API_call("#{$canvasUrl}/api/v1/sections/#{sectionValue}",
-										                               $canvasToken)
-										if (section_data.has_key?("sis_section_id"))
-											sectionParsedSISID=section_data["sis_section_id"]
-											if (sectionParsedSISID != nil)
-												## for now we will use just the last 5-digit of the section id
-												sectionParsedSISID = sectionParsedSISID[4,8]
-
-												result_json = setMPathwayUrl($canvasUrl, $esbUrl, esbToken, sisTermId, sectionParsedSISID, courseId)
-												message = "set url result for course id=#{sectionParsedSISID} with Canvas courseId=#{courseId}: result status=#{result_json["setLMSURLResponse"]["Resultcode"]} and result message=#{result_json["setLMSURLResponse"]["ResultMessage"]}"
-												p message
-												# write into output file
-												outputFile.write(message)
-											end
-										end
+									if (sectionKey=="sis_section_id")
+										## get the sis_section_id value
+										sectionParsedSISID=sectionValue
+										break
+									end
+								end
+								if (sectionParsedSISID != nil)
+									## sis_section_id is 9-digit: <4-digit term id><5-digit section id>
+									# we will use just the last 5-digit of the section id
+									sectionParsedSISID = sectionParsedSISID[4,8]
+									result_json = setMPathwayUrl($canvasUrl, $esbUrl, esbToken, sisTermId, sectionParsedSISID, courseId)
+									if (result_json.has_key? "setLMSURLResponse")
+										message = "set url result for section id=#{sectionParsedSISID} with Canvas courseId=#{courseId}: result status=#{result_json["setLMSURLResponse"]["Resultcode"]} and result message=#{result_json["setLMSURLResponse"]["ResultMessage"]}"
+									else
+										message = "set url result for section id=#{sectionParsedSISID} with Canvas courseId=#{courseId}: result #{result_json.to_s}"
 									end
+									# write into output file
+									outputFile.write(message + "\n")
 								end
 							}
 						end
```

Comment:
My main concerns are:
- Hard to follow because of nested if clauses and loops.
- Error handling is unclear because of being hard to follow, and because of a lack of rescue clauses.  E.g. what happens if there is a json parse error?  Are all updates lost? Just some?  Just one?  How can it be restarted to pick up after an error?